deps/wrapper.cc

```diff
@@ -82,13 +84,17 @@ void initialize_coreworker_worker(
             const std::string name_of_concurrency_group_to_execute,
             bool is_reattempt,
             bool is_streaming_generator) {
-            RAY_LOG(DEBUG) << "ray_core_worker_julia_jll: entered task_execuation_callback...";
-          // task_executor(ray_function, returns, args);
-          int pid = task_executor(ray_function, args);
-          std::string str = std::to_string(pid);
-          auto memory_buffer = std::make_shared<LocalMemoryBuffer>(reinterpret_cast<uint8_t *>(&str[0]), str.size(), true);
+
+          std::vector<std::shared_ptr<LocalMemoryBuffer>> return_vec;
+          auto args_ptr = static_cast<const void *>(&args);
+          task_executor(ray_function, &return_vec, args_ptr);
+
+          RAY_CHECK(return_vec.size() == 1);
           RAY_CHECK(returns->size() == 1);
-          (*returns)[0].second = std::make_shared<RayObject>(memory_buffer, nullptr, std::vector<rpc::ObjectReference>());
+
+          // TODO: support multiple return values
+          std::shared_ptr<LocalMemoryBuffer> buffer = return_vec[0];
+          (*returns)[0].second = std::make_shared<RayObject>(buffer, nullptr, std::vector<rpc::ObjectReference>(), true);
```

Comment:
Can we get away with using `copy=false`?
```suggestion
          (*returns)[0].second = std::make_shared<RayObject>(buffer, nullptr, std::vector<rpc::ObjectReference>(), false);
```

Reply:
worth a shot

Reply:
worked